runtime/v2/shim/reaper_unix.go

```diff
@@ -41,13 +42,20 @@ func Reap() error {
 	Default.Lock()
 	for c := range Default.subscribers {
 		for _, e := range exits {
-			c <- runc.Exit{
+			select {
+			case c <- runc.Exit{
```

Comment:
We don't want to discard in v2 either right?

Reply:
I hadn’t noticed until cherry picking that @Random-Liu only reverted the v1 discard. But I assumed if we want to revert the other we need a new PR in master, and then we can cherry pick that to `release/1.2` as well

Reply:
Ok, i'll do that now